frontend/src/models/rewards.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from '@common'
+import { RewardsScheduleSegment, ParticipantStudy } from '@api'
+import { useEnv } from './environment'
+import { sortBy, last } from 'lodash-es'
+
+export interface CalculatedRewardsScheduleSegment extends RewardsScheduleSegment {
+    totalPoints: number
+    index: number
+}
+
+export function rewardPointsEarned(studies: ParticipantStudy[]): number {
+    return studies.reduce((points, study) => {
+        if (study.participationPoints && study.completedAt) {
+            points += study.participationPoints
```

Comment:
yes, mutating doesn't make sense since it's using reduce, it just needs to return the value.  I do like an if vs the ternary though since we shouldn't do anything if it's not completed.